SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m

```diff
@@ -17,6 +17,510 @@ static void FreeImageData(void *info, const void *data, size_t size) {
+        static CGColorSpaceRef bt2020pq = NULL;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            if (@available(macOS 10.14.6, iOS 13.0, tvOS 13.0, *)) {
+                bt2020pq = CGColorSpaceCreateWithName(kCGColorSpaceITUR_2020_PQ_EOTF);
+            } else {
+                bt2020pq = defaultColorSpace;
+            }
+        });
+        *ref = bt2020pq;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(colorPrimaries == AVIF_NCLX_COLOUR_PRIMARIES_P3 &&
+       transferCharacteristics == AVIF_NCLX_TRANSFER_CHARACTERISTICS_SRGB) {
+        static CGColorSpaceRef p3 = NULL;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            if (@available(macOS 10.11.2, iOS 9.3, tvOS 9.3, *)) {
+                p3 = CGColorSpaceCreateWithName(kCGColorSpaceDisplayP3);
+            } else {
+                p3 = defaultColorSpace;
+            }
+        });
+        *ref = p3;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(colorPrimaries == AVIF_NCLX_COLOUR_PRIMARIES_P3 &&
+       transferCharacteristics == AVIF_NCLX_TRANSFER_CHARACTERISTICS_BT2100_HLG) {
+        static CGColorSpaceRef p3hlg = NULL;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            if (@available(macOS 10.14.6, iOS 13.0, tvOS 13.0, *)) {
+                p3hlg = CGColorSpaceCreateWithName(kCGColorSpaceDisplayP3_HLG);
+            } else {
+                p3hlg = defaultColorSpace;
+            }
+        });
+
+        *ref = p3hlg;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(colorPrimaries == AVIF_NCLX_COLOUR_PRIMARIES_P3 &&
+       transferCharacteristics == AVIF_NCLX_TRANSFER_CHARACTERISTICS_BT2100_PQ) {
+        static CGColorSpaceRef p3pq = NULL;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            if (@available(macOS 10.14.6, iOS 13.0, tvOS 13.0, *)) {
+                p3pq = CGColorSpaceCreateWithName(kCGColorSpaceDisplayP3_PQ_EOTF);
+            } else {
+                p3pq = defaultColorSpace;
+            }
+        });
+        *ref = p3pq;
+        *shouldRelease = FALSE;
+        return;
+    }
+    if(colorPrimaries == AVIF_NCLX_COLOUR_PRIMARIES_P3 &&
+       transferCharacteristics == AVIF_NCLX_TRANSFER_CHARACTERISTICS_LINEAR) {
+        static CGColorSpaceRef p3linear = NULL;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+            if (@available(macOS 10.14.3, iOS 12.3, tvOS 12.3, *)) {
+                p3linear = CGColorSpaceCreateWithName(kCGColorSpaceExtendedLinearDisplayP3);
+            } else {
+                p3linear = defaultColorSpace;
+            }
+        });
+        *ref = p3linear;
+        *shouldRelease = FALSE;
+        return;
+    }
+
+    *ref = CreateColorSpaceRGB(colorPrimaries, transferCharacteristics);
```

Comment:
This is the fallback case for CGColorSpace which does not supported ?

Reply:
This branch is for when there is no corresponding built-in color space in `CGColorSpace.h`, like `kCGColorSpaceSRGB`.
If there is better way to create color spaces, please tell me.

Reply:
OK. Seems we need to refactory soon....There are so many calc code and utils C function right now :). Don't need to do this right now.

Reply:
OK. I removed the code that creates CGColorSpace using vImage, and left a TODO comment about it.

Yeah, I also felt we need refactoring.

Reply:
Sorry, I mis-understand your message. I reverted https://github.com/SDWebImage/SDWebImageAVIFCoder/pull/13/commits/71ecfa1d0a2273d52f9042199221708297da319a .